In [ ]:
frequency_urls = {
    "small_cell_carcinoma_icdom": "https://progenetix.org/services/intervalFrequencies/?output=pgxseg&filters=pgx:icdom-80413" 
}

frequency_tables = {}

for source_query, url in frequency_urls.items():
    print(f"Downloading frequency table for {source_query}")
    response = requests.get(url, timeout=120)
    response.raise_for_status()
    df_freq = pd.read_csv(StringIO(response.text), sep="\t", comment="#")
    df_freq["source_query"] = source_query
    frequency_tables[source_query] = df_freq

for source_query, df_freq in frequency_tables.items():
    print(source_query, df_freq.shape)

    frequency_tables["small_cell_carcinoma_icdom"].head()
   


small_cell_carcinoma_icdom (3106, 7)


In [15]:
 frequency_tables["small_cell_carcinoma_icdom"].columns

Index(['reference_name', 'start', 'end', 'gain_frequency', 'loss_frequency',
       'no', 'source_query'],
      dtype='str')

In [11]:
import requests
import pandas as pd

import matplotlib.pyplot as plt
from io import StringIO

url = "https://progenetix.org/beacon/biosamples"

# Define the query parameters.
# "filters" is used to restrict the query.
# "limit" controls how many results we want to retrieve.

params = {
    "filters": "NCIT:C4872",
    "limit": 5
}

response = requests.get(url, params=params)

# Show the HTTP status code.
# A value of 200 usually means the request was successful.

response.status_code

200

Answer to question: 
Inspect shape: 3106, 7 (rows (samples), columns (features))
Inspect columns: ['reference_name', 'start', 'end', 'gain_frequency', 'loss_frequency',
       'no', 'source_query'],
      dtype='str'. 

Explain why the frequency table has a different shape from the sample table.
    - what one row represents in the sample table
    - what one row represents in the frequency table
    - why these two tables cannot be merged row by row

The frequency table has a different shape from the sample table because they describe the same data in 2 different levels of detail.  In the sample table each row is one biological sample vs in a frequency table, a row is one category (a label value) with its count (how often it occurs). So the frequency table is a compressed representation of the sample table. Essentially, the sample table is raw data (many rows, detailed) vs the frequency table is summarized data (fewer rows, aggregated). 
They are incompatible for row-wise merging because one row in the frequency table represents multiple rows in the sample table. The frequency table doesn't have individual identity information which the sample table has. 
